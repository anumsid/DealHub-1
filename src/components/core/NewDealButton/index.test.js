import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { NewDealButton } from './index'

describe('New Deal Button Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      authToken: 'xxx',
      location: { pathname: '' },
      onClick: onClick,
    }
    wrapper = shallow(<NewDealButton {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Dispatches click when clicked', () => {
    wrapper.find('Button').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })
})
