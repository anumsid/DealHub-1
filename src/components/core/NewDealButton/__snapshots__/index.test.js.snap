// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Deal Button Component Matches snapshot 1`] = `
<Button
  as="button"
  className="new-deal-button"
  onClick={[Function]}
  secondary={true}
>
  Add New Deal
</Button>
`;
